docs/src/catalyst_applications/optimization_ode_param_fitting.md

```diff
@@ -0,0 +1,168 @@
+- `save_idxs=4`: The measured species (*P*) is the 4th species in our species vector (`species(rn)`). To ensure that the concentration of the right species is evaluated against the data, we set the numeric integrator to only save the value of this species.
+
+Now we can create an `OptimizationProblem` using our `loss_function` and some initial guess of parameter values from which the optimiser will start:
+```@example diffeq_param_estim_1 
+optprob = OptimizationProblem(loss_function, [1.0, 1.0, 1.0])
+nothing #
+```
+
+!!! note
+    `OptimizationProblem` cannot currently accept parameter values in the form of a map (e.g. `[:kB => 1.0, :kD => 1.0, :kP => 1.0]`). These must be provided as individual values (using the same order as the parameters occur in in the `parameters(rs)` vector).
+
+Finally, we can optimise `optprob` to find the parameter set that best fits our data. Optimization.jl does not provide any optimisation methods by default. Instead, for each supported optimisation package, it provides a corresponding wrapper-package to import that optimisation package for using with Optimization. E.g., if we wish to [Optim.jl](https://github.com/JuliaNLSolvers/Optim.jl)'s [Nelder-Mead](https://en.wikipedia.org/wiki/Nelder%E2%80%93Mead_method) method, we must install and import the OptimizationOptimJL package. A summary of all, by Optimization.jl, supported optimisation packages can be found [here](https://docs.sciml.ai/Optimization/stable/#Overview-of-the-Optimizers). Here, we import the Optim.jl package and uses it to minimise our cost function (thus finding a parameter set that fits the data):
+```@example diffeq_param_estim_1 
+using OptimizationOptimJL
+optsol = solve(optprob, NelderMead())
+nothing # hide
+```
+
+We can now simulate our model for the corresponding parameter set, checking that it fits our data.
+```@example diffeq_param_estim_1 
+oprob_fitted = remake(oprob; p=optsol.u)
+fitted_sol = solve(oprob_fitted, Tsit5())
+plot!(fitted_sol; idxs=:P, label="Fitted solution", linestyle=:dash, lw=6, color=:lightblue)
+```
+
+!!! note
+Here, a good exercise is to check the resulting parameter set and note that, while it creates a good fit to the data, it does not actually correspond to the original parameter set. [Identifiability](https://www.sciencedirect.com/science/article/pii/S1364815218307278) is a concept that studies how to deal with this problem.
+
+Say that we instead would like to use the [Broyden–Fletcher–Goldfarb–Shannon](https://en.wikipedia.org/wiki/Broyden%E2%80%93Fletcher%E2%80%93Goldfarb%E2%80%93Shanno_algorithm) algorithm, as implemented by the [NLopt.jl](https://github.com/JuliaOpt/NLopt.jl) package. In this case we would run:
+```@example diffeq_param_estim_1 
+using OptimizationNLopt
+sol = solve(optprob, BFGS())
```

Comment:
```suggestion
sol = solve(optprob, NLopt.LD_LBFGS())
```

Reply:
For this example, do you have a suggestion of a method that I can reach without doing `packagename.optimiser`. Alternatively, have a second one used this way. I want to keep things simple, and it feels weird accessing the two optimisers (currently `NelderMead` and `BFGS`) using two different interfaces.

Reply:
You can change the NelderMEad to Optim.NelderMead that'll still work

Reply:
Sounds perfect.

Why does some methods need to be accessed like `packagename.optimiser` ad some can be accessed directly?